contracts/SafeProtocolRegistry.sol

```diff
@@ -59,8 +59,8 @@ contract SafeProtocolRegistry is ISafeProtocolRegistry, Ownable2Step {
         ModuleInfo memory moduleInfo = listedModules[module];
 
         // Check if module is already listed or if moduleTypes is greater than 8.
-        // Maximum allowed value of moduleTypes is 7. i.e. 2^0 (Plugin) + 2^1 (Function Handler) + 2^2 (Hooks)
-        if (moduleInfo.listedAt != 0 || moduleTypes > 7) {
+        // Maximum allowed value of moduleTypes is 31. i.e. 2^0 (Plugin) + 2^1 (Function Handler) + 2^2 (Hooks) + 2^3 (Signature Validator hooks) + 2^4 (Signature Validator)
+        if (moduleInfo.listedAt != 0 || moduleTypes > 31) {
```

Comment:
Would it be possible to simplify this method (code style wise).

I.e. create a function `optionalCheckIncterfaceSupport`:

```
function optionalCheckIncterfaceSupport(address module, uint8 moduleTypes, uint8 moduleTypeToCheck, uint8 interfaceId) internal {
if (
            moduleTypes & moduleTypeToCheck == moduleTypeToCheck && !IERC165(module).supportsInterface(interfaceId)
        ) {
            revert ModuleDoesNotSupportExpectedInterfaceId(module, interfaceId);
        }
}
```

And then we can call this 4 times. 

Would be interesting to know if this would reduce code size :thinking:

Reply:
Old implementation:
`SafeProtocolRegistry 5862 bytes (limit is 24576)`

With the commit https://github.com/safe-global/safe-core-protocol/pull/109/commits/dfb0e0cc3fd9a1c481f6cad90b03adbbb4150bcb:
`SafeProtocolRegistry 4923 bytes (limit is 24576)`